### PR TITLE
Add gpu simulation support to client

### PR DESCRIPTION
**Pull request by paaige**

### CODE REVIEW

File: general_superstaq/typing.py [hunk 1]
```diff
@@ -54,5 +54,6 @@
         "data": Optional[Dict[str, Any]],
         "samples": Optional[Counter],
         "shots": Optional[int],
+        "result": Optional[Counter],
```
Comment (vtomole) [inline general_superstaq/typing.py:57]: Why are we adding this? Why can't we use `samples` here instead?

**Review by vtomole — CHANGES_REQUESTED** [review]
Need to pass the checks.

**Review by vtomole — COMMENTED** [review]
We need to add a `method` argument after https://github.com/SupertechLabs/SuperstaQ/pull/1500#pullrequestreview-1082958740

**Review by vtomole — APPROVED** [review]
LGTM

**Review by vtomole — COMMENTED** [review]
Wait. Is options a string or a dict? https://github.com/SupertechLabs/cirq-superstaq/pull/312/files#r955130631

**Comment by paaige** [comment]
It's a JSON serialized dictionary (a string)

**Pull request merged**